src/Form/FormFieldControl.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useFormFieldContext } from './FormFieldContext';
+import { FormControl } from '..';
+import { useHasValue, callAllHandlers } from './fieldUtils';
+
+const FormFieldControl = React.forwardRef(({ className, ...props }, ref) => {
+  const {
+    size,
+    id,
+    isInvalid,
+    isValid,
+    onFocus,
+    onBlur,
+    placeholder,
+    'aria-describedby': ariaDescribedBy,
+  } = useFormFieldContext(props);
+
+  const [hasValue, checkInputEventValue] = useHasValue({ defaultValue: props.defaultValue, value: props.value });
+
+  return (
+    <FormControl
+      {...props}
+      ref={ref}
+      size={size}
+      id={id}
+      isInvalid={isInvalid}
+      isValid={isValid}
```

Comment:
I'm skeptical of something that has `isInvalid` and `isValid` states. What happens if they are both true or both false at the same time? (And/or can they be renamed to be more descriptive about what they are referring to?

Reply:
I agree with you, but this props api matched the current FormControl from react bootstrap, making this backward compatible. I think we should keep it as is for now. Re: `(And/or can they be renamed to be more descriptive about what they are referring to?` what do you mean?